src/App/Pages/Vault/GroupingsPage/GroupingsPageViewModel.cs

```diff
@@ -439,12 +502,27 @@ private async Task LoadDataAsync()
 
             if (MainPage)
             {
-                Folders = await _folderService.GetAllDecryptedAsync();
-                NestedFolders = await _folderService.GetAllNestedAsync();
+                var decFolders = await _folderService.GetAllDecryptedAsync();
+                var decCollections = await _collectionService.GetAllDecryptedAsync();
+                if (IsVaultFilterMyVault)
+                {
+                    Folders = BuildFolders(decFolders);
+                    Collections = null;
+                }
+                else if (IsVaultFilterOrgVault && !string.IsNullOrWhiteSpace(orgId))
+                {
+                    Folders = BuildFolders(decFolders);
+                    Collections = decCollections?.Where(c => c.OrganizationId == orgId).ToList();
+                }
+                else
+                {
+                    Folders = decFolders;
+                    Collections = decCollections;
+                }
+                NestedFolders = await _folderService.GetAllNestedAsync(Folders);
                 HasFolders = NestedFolders.Any(f => f.Node?.Id != null);
-                Collections = await _collectionService.GetAllDecryptedAsync();
-                NestedCollections = await _collectionService.GetAllNestedAsync(Collections);
-                HasCollections = NestedCollections.Any();
+                NestedCollections = Collections != null ? await _collectionService.GetAllNestedAsync(Collections) : null;
+                HasCollections = NestedCollections?.Any() ?? false;
```

Comment:
Can we extract it to another method? Same reason as before: `LoadDataAsync` is IMO already too large, it could be something like:
`async Task FillFoldersAndCollectionsAsync();`